Kommunicate/Classes/KMConversationListViewController.swift

```diff
@@ -0,0 +1,571 @@
+//
+//  KMConversationListViewController.swift
+//  Kommunicate
+//
+//  Created by Sunil on 28/01/20.
+//
+
+import Foundation
+import ApplozicSwift
+import Applozic
+
+public class KMConversationListViewController : ALKBaseViewController, Localizable {
+    public var conversationViewController: KMConversationViewController?
+    public var conversationViewModelType = ALKConversationViewModel.self
+    public var conversationListTableViewController: ALKConversationListTableViewController
+
+    var searchController: UISearchController!
+    var searchBar: CustomSearchBar!
+    lazy var resultVC = ALKSearchResultViewController(configuration: configuration)
+
+    public var dbService = ALMessageDBService()
+    public var viewModel = ALKConversationListViewModel()
+
+    // To check if coming from push notification
+    var contactId: String?
+    var channelKey: NSNumber?
+    var conversationId: NSNumber?
+
+    var tableView: UITableView
+
+    lazy var rightBarButtonItem: UIBarButtonItem = {
+        let icon = UIImage(named: "fill_214", in: Bundle(for: ALKConversationListViewController.self), compatibleWith: nil)
+        let barButton = UIBarButtonItem(
+            image: icon,
+            style: .plain,
+            target: self, action: #selector(compose)
+        )
+        return barButton
+    }()
+
+    fileprivate var tapToDismiss: UITapGestureRecognizer!
+    fileprivate var alMqttConversationService: ALMQTTConversationService!
+    fileprivate let activityIndicator = UIActivityIndicatorView(style: UIActivityIndicatorView.Style.gray)
+    fileprivate var localizedStringFileName: String!
+
+    public required init(configuration: ALKConfiguration) {
+        conversationListTableViewController = ALKConversationListTableViewController(
+            viewModel: viewModel,
+            dbService: dbService,
+            configuration: configuration,
+            showSearch: false
+        )
+        tableView = conversationListTableViewController.tableView
+        super.init(configuration: configuration)
+        conversationListTableViewController.delegate = self
+        localizedStringFileName = configuration.localizedStringFileName
+    }
+
+    public required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        removeObserver()
```

Comment:
We can `override` baseVCs `addObserver` and `removeObserver` method. Then we don't have to call these methods in this class.